### PR TITLE
[SparkR][SPARK-21381]:SparkR: pass on setHandleInvalid for classification algorithms

### PR DESCRIPTION
**Pull request by wangmiao1981**
## What changes were proposed in this pull request?

SPARK-20307 Added handleInvalid option to RFormula for tree-based classification algorithms. We should add this parameter for other classification algorithms in SparkR.

This is a followup PR for SPARK-20307.

## How was this patch tested?

New Unit tests are added.

### CODE REVIEW

**Comment by wangmiao1981** [comment]
@felixcheung This is a follow-up PR of JIRA-20307.

**Comment by wangmiao1981** [comment]
Trigger windows check.

**Comment by wangmiao1981** [comment]
Reopen for windows check

**Review by felixcheung — COMMENTED** [review]
could you update this to make it consistent with the earlier PR? I think it's mostly the param document wording

**Comment by wangmiao1981** [comment]
Sure. I am reading the #18613 comments. Just come back from a business travel. Thanks!

**Comment by wangmiao1981** [comment]
@yanboliang after #18613, unit tests fails if "skip" is used.

For example,
data <- data.frame(clicked = base::sample(c(0, 1), 10, replace = TRUE),
  someString = base::sample(c("this", "that"), 10, replace = TRUE),
                            stringsAsFactors = FALSE)
  trainidxs <- base::sample(nrow(data), nrow(data) * 0.7)
  traindf <- as.DataFrame(data[trainidxs, ])
  testdf <- as.DataFrame(rbind(data[-trainidxs, ], c(0, "the other")))
  model <- spark.mlp(traindf, clicked ~ ., layers = c(1, 3), handleInvalid = "keep")
  predictions <- predict(model, testdf)
  expect_equal(class(collect(predictions)$clicked[1]), "character")

It fails the as if "error" is used.

If I change "skip" to "keep", then the predictions$click[0] is NULL.
> collect(predictions)
[1] clicked    someString prediction
<0 rows> (or 0-length row.names)
> collect(predictions)$click[1]
[[1]]
NULL

I am not sure whether this is expected or there is a bug.

Before, the units work fine.

**Comment by yanboliang** [comment]
@wangmiao1981 This is expected, see my comment [here](https://github.com/apache/spark/pull/18613#discussion_r127011929) . This uncovers an existing bug for ```forceIndexLabel```. I will send a fix later, please use ```keep``` to test ```handleInvalid``` currently. Thanks.

**Comment by wangmiao1981** [comment]
@yanboliang Thanks for your reply! I will change the unit tests now.

**Comment by wangmiao1981** [comment]
@yanboliang I have made changes accordingly. Thanks!

**Comment by wangmiao1981** [comment]
@felixcheung Can you take a look? Thanks!

**Comment by felixcheung** [comment]
I'll take a look

**Review by felixcheung — COMMENTED** [review]
sorry about the delay, one comment otherwise LG

File: R/pkg/R/mllib_tree.R [hunk 1]
```diff
@@ -374,10 +382,11 @@ setMethod("write.ml", signature(object = "GBTClassificationModel", path = "chara
 #'                     nodes. If TRUE, the algorithm will cache node IDs for each instance. Caching
 #'                     can speed up training of deeper trees. Users can set how often should the
 #'                     cache be checkpointed or disable it by setting checkpointInterval.
-#' @param handleInvalid How to handle invalid data (unseen labels or NULL values) in classification model.
-#'        Supported options: "skip" (filter out rows with invalid data),
-#'                           "error" (throw an error), "keep" (put invalid data in a special additional
-#'                           bucket, at index numLabels). Default is "error".
+#' @param handleInvalid How to handle invalid data (unseen labels or NULL values) in features and label
+#'                      column of string type.
```
Comment (felixcheung) [inline R/pkg/R/mllib_tree.R]: this was only for classification though, so the original text `in classification model.` we should keep.
ditto for decisionTree and gbt in this .R file

**Review by felixcheung — APPROVED** [review]

**Comment by felixcheung** [comment]
merged to master

**Pull request closed without merging**